### ISSUE TITLE
transorfmer model  dropout mistake

### ISSUE THREAD

**Issue by tanglifutlf** [OpenNMT/OpenNMT-tf#207]
outputs = tf.layers.dropout(
      outputs,
      rate=dropout,
training=mode == tf.estimator.ModeKeys.TRAIN)

this is the code in path "opennmt/layers/transformer.py"   ,reveletn line is 342 to 345,   in function "tf.layers.dropout"  the data  "output" should be "inputs"

**Comment by guillaumekln** [comment]
No, the code is correct.

Quoting the paper:

> We apply dropout to the output of each sub-layer, before it is added to the sub-layer input and normalized.

In the code:

* `outputs` refers to the sub-layer output
* `inputs` refers to the sub-layer input
